src/cascadia/TerminalApp/App.cpp

```diff
@@ -899,12 +899,22 @@ namespace winrt::TerminalApp::implementation
     void App::_UpdateTitle(std::shared_ptr<Tab> tab)
     {
         auto newTabTitle = tab->GetFocusedTitle();
-        tab->SetTabText(newTabTitle);
+        const auto lastFocusedProfile = tab->GetFocusedProfile().value();
+        const auto* const matchingProfile = _settings->FindProfile(lastFocusedProfile);
+
+        const auto suppressApplicationTitle = matchingProfile->GetSuppressApplicationTitle();

```

Comment:
You need a null check here: since `matchingProfile` is a pointer type, it could be `nullptr`. That'd indicate that the profile was _removed from settings_ (!) before this happened.